app/src/main/java/com/android/sample/ui/Gallery.kt

```diff
@@ -0,0 +1,115 @@
+package com.android.sample.ui
+
+import android.net.Uri
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.Image
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import coil.compose.rememberAsyncImagePainter
+import coil.request.ImageRequest
+
+@Composable
+fun ImagePicker(onImagePicked: (Uri?) -> Unit, buttonText: String = "Select Image") {
+  // Launches the gallery picker
+  val pickImageLauncher =
+      rememberLauncherForActivityResult(ActivityResultContracts.GetContent()) { uri ->
+        onImagePicked(uri) // Pass the selected URI back
+      }
+  Button(onClick = { pickImageLauncher.launch("image/*") }, Modifier.testTag("uploadPicture")) {
+    Text(buttonText)
+  }
+}
+
```

Comment:
Handling null or empty URLs: Nice implementation of conditional URI parsing to handle empty or null strings, which reduces unnecessary API calls or load attempts.